SPHINXsys/src/shared/common/image_mhd.h

```diff
@@ -0,0 +1,165 @@
+/* -------------------------------------------------------------------------*
+*								SPHinXsys									*
+* --------------------------------------------------------------------------*
+* SPHinXsys (pronunciation: s'finksis) is an acronym from Smoothed Particle	*
+* Hydrodynamics for industrial compleX systems. It provides C++ APIs for	*
+* physical accurate simulation and aims to model coupled industrial dynamic *
+* systems including fluid, solid, multi-body dynamics and beyond with SPH	*
+* (smoothed particle hydrodynamics), a meshless computational method using	*
+* particle discretization.													*
+*																			*
+* SPHinXsys is partially funded by German Research Foundation				*
+* (Deutsche Forschungsgemeinschaft) DFG HU1527/6-1, HU1527/10-1				*
+* and HU1527/12-1.															*
+*                                                                           *
+* Portions copyright (c) 2017-2020 Technical University of Munich and		*
+* the authors' affiliations.												*
+*                                                                           *
+* Licensed under the Apache License, Version 2.0 (the "License"); you may   *
+* not use this file except in compliance with the License. You may obtain a *
+* copy of the License at http://www.apache.org/licenses/LICENSE-2.0.        *
+*                                                                           *
+* --------------------------------------------------------------------------*/
+/**
+* @file image_mesh_shape.h
+* @brief x
+* @details x
+*			x
+* @author	Yijin Mao
+*/
+#ifndef IMAGE_MHD_H
+#define IMAGE_MHD_H
+
+#include "small_vectors.h"
+#include "sph_data_containers.h"
+
+#include <iostream>
+#include <string>
+#include <fstream>
+
+namespace SPH {
+
+	enum Image_Data_Type
+	{
+		MET_FLOAT,
+		MET_UCHAR,
+		MET_LONG
+	};
+
+	enum Output_Mode
+	{
+		BINARY,
+		ASCII
+	};
+
+	template <typename T, int nDims>
+	class ImageMHD
+	{
+	public:
+		ImageMHD() {};
+		// constructor for input files
+		ImageMHD(std::string full_path_file);
+		// constructor for sphere 
+		ImageMHD(Real radius, Vec3i dxdydz, Vec3d spacings);
+		//ImageMHD(Vec3i dxdydz, Vec3d spacings, T*data);
+		~ImageMHD();
+
+		void set_objectType(std::string objectType) 
```

Comment:
I found this is good way to  access/set function member. I have seen many direct access to members in class and members are not protected in Sphinxsys

Reply:
I agree. Usually direct access to class member is not recommended. @Xiangyu-Hu @chuvirtonomy